base.sh

```diff
@@ -132,13 +130,24 @@ stcountmax=5
 chkslint=60
 chksltot=%(job_time)d
 
-chksave=%(chkpt_save_opt)s
-
-cr_restart_restore="--no-restore-pid"
-
 chkprestage="$chkdir/prestage"
 chkpoststage="$chkdir/poststage"
 
+timestamp_latest_checkpoint() {
+    # determine timestamp of most recent checkpoint (0 if no checkpoint files are found)
+    tmpfile=`mktemp`
+    find $chkdir -name '*.dmtcp' > $tmpfile 2> /dev/null
```

Comment:
something like
```
timestamp=$(ls -t $chkdir/*.dmtcp |head -1 | xargs -i '{}' stat -c %%Y '{}')
echo ${timestamp:-0}
```
the stat command might require `$chkdir/{}`...

Reply:
Don't parse the output of `ls` if possible. Using GNU find you can print the modification time:

```bash
$ find . -type f -printf '%T@\n' | sort -n | tail -1
1505934062.0000000000
# trim decimals
$ find . -type f -printf '%T@\n' | sort -n | tail -1 | cut -f1 -d'.'
1505934062
```

Reply:
:+1:, and maybe add `-maxdepth 0` to the find command (just in case)